pkg/status/status.go

```diff
@@ -225,14 +225,11 @@ func (r *reporter) monitorClusterStatus() {
 			}
 
 			// Report that marketplace is available
-			if cohelpers.IsStatusConditionFalse(r.clusterOperator.Status.Conditions, configv1.OperatorAvailable) {
-				reason := "OperatorAvailable"
-				conditionListBuilder := clusterStatusListBuilder()
-				conditionListBuilder(configv1.OperatorProgressing, configv1.ConditionFalse, fmt.Sprintf("Successfully progressed to release version: %s", r.version), reason)
-				statusConditions := conditionListBuilder(configv1.OperatorAvailable, configv1.ConditionTrue, fmt.Sprintf("Available release version: %s", r.version), reason)
-				statusErr = r.setStatus(statusConditions)
-				break
-			}
+			reason := "OperatorAvailable"
```

Comment:
why not just create it in the succeeded state to start with? all this really confirms is that the our `timer` works, right? (which seems like a proxy for "is the pod alive" which we'll already have a signal for)

Reply:
+1

It feels like this whole reporting mechanism should be ripped out and we can just have a function that does this.

Reply:
@ecordell @kevinrizza if we rip out the whole thing, we will essentially rip out the ability to report status based on the availability of the default catalogsources (the way we used to with default CatalogSources). 

Are we saying that we don't forsee a requirement coming in that'll ask that of marketplace ever?

Reply:
Also, we still want a thread to keep checking if the operator is up, and if not[ report `OperatorExited` ](https://github.com/operator-framework/operator-marketplace/blob/master/pkg/status/status.go#L190 )right?

Reply:
> all this really confirms is that the our timer works, right? 

This thread running an infinite loop actually does three things: 
1) On startup, creates the clusteroperator and sets the message to `Determining Status` and condition to `Available:False` `Progressing: True` `Degraded: False`
2) Once the startup is done it sets the condition to `Available:True` `Progressing: False` `Degraded: False`
3) If the deployment is deleted it sets the clusteroperator condition to `Available:False` with `Reason: OperatorExited`.

I'm guessing we want the deployment to be monitored and the clusteroperator status to be set as `Available:False` if the deployment is deleted.

Reply:
The current loop is really :

1) On startup, creates the clusteroperator and sets the message to `Determining Status` and condition to `Available:False` `Progressing: True` `Degraded: False`
2) 20 seconds later, regardless of what else has happened, sets `Available:True` `Progressing: False` `Degraded: False`
3) If the deployment is deleted it sets the clusteroperator condition to `Available:False` with `Reason: OperatorExited`.

I'm just saying that if there's nothing (aside from the stopchannel) that really prevents us from going available anymore, let's just make this:

1) On startup, creates the clusteroperator and sets the message to Available:True` `Progressing: False` `Degraded: False`
2) If the deployment is deleted it sets the clusteroperator condition to `Available:False` with `Reason: OperatorExited`.

>  where do we have the signal for that?

I just mean that the pod will be failing and there are already cluster alerts for failing pods.

Reply:
At the time that this was created, there was an e2e test which ensured that operators don't immediately report available. This was over a year ago and might have changed.

Reply:
I reached out to Trevor King and this test does not seem to exist anymore.